pkg/cache/cache.go

```diff
@@ -220,18 +220,35 @@ func (c *Cache) WaitForPodsReady(ctx context.Context) {
 	defer c.Unlock()
 
 	log := ctrl.LoggerFrom(ctx)
+
+	podsReadyCond := func(ch chan any) {
+		c.Lock()
+		defer c.Unlock()
+		defer close(ch)
+		c.podsReadyCond.Wait()
+		ch <- struct{}{}
+	}
+
+	podsReadyCondCh := make(chan any)
+
 	for {
 		if c.podsReadyForAllAdmittedWorkloads(ctx) {
```

Comment:
it doesn't look like this function needs the `ctx`. We can pass a the `log` instead.
You can do it in this PR.